staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -762,25 +786,62 @@ func (s *Controller) nodeSyncInternal(ctx context.Context, workers int) {
 }
 
 // nodeSyncService syncs the nodes for one load balancer type service
-func (s *Controller) nodeSyncService(svc *v1.Service) bool {
+func (s *Controller) nodeSyncService(svc *v1.Service, oldNodes, newNodes []*v1.Node) bool {
```

Comment:
I changed the signature to this because I wanted to have a synchronized retrieval of `nodes`, and update of `lastSyncedNodes` in `updateLoadBalancerHosts`. `nodeSyncService` is parallelized per `workers` and I didn't want to run the risk of having different workers executing `nodeSyncService` with a different set of nodes - which might happen if we list nodes internally. 

I can comment on the assumptions.